media/windows/ricoh-printer/ricoh_db_create.py

```diff
@@ -0,0 +1,121 @@
+#!/user/bin/env python3
+
+# Script which runs daily to update staff print totals
+# based on CSVs generated by the printer itself.
+
+import sys
+sys.path.insert(1, '../../python/')
+
+import os
+import re
+import csv
+import datetime
+import sqlite3
+from sqlite3 import Error
+
+#import ECC
+#import PDSChurch
+
+from pprint import pprint
+from pprint import pformat
+
+###########################################################
+
+ricoh_db = "media/windows/ricoh-printer/ricoh.db"
+today = datetime.date.today()
+datestring = f'{today.year}{today.month}{today.day}'
+debug_datestring = '20201013'
+
+def _strip(value):
+    #The CSV puts brackets around values, which we don't want
+    return value.strip('[').strip(']')    
+
+def load_latest_csv():
+    
+    filename = f'RICOH IM C4500_usercounter_{datestring}.csv'
+    
+    debug_filename = 'media/windows/ricoh-printer/RICOH IM C4500_usercounter_20201013.csv'
+    
+    csv_rows = list()
+
+    with open(debug_filename, encoding='utf-8') as csvfile: #TODO: change to filename when retrieval is automated
+        csvreader = csv.DictReader(csvfile)
+        for row in csvreader:
+            csv_rows.append(row)
+
+    print(f"== Loaded {len(csv_rows)} staffers from latest CSV")
+    return csv_rows
+
+def extract_csv_data(csv_rows):
+
+    def _extract_staffer(row, num, name):
+        return {
+            'num'           : num,
+            'name'          : name,
+            'total'         : int(row['Total Prints'].strip()),
+            'b&wtotal'      : int(row['B & W(Total Prints)'].strip()),
+            'colortotal'    : int(row['Color(Total Prints)'].strip()),
+        }
+
+    csv_staffers = list()
+    
+    for row in csv_rows:
+        try:
+            num = int(_strip(row['User']))
+            name = _strip(row['Name'])
+        except ValueError as e:
+            print(e)
+            num = 000
+            name = 'Other'
+        this_staffer = _extract_staffer(row, num, name)
+        csv_staffers.append(this_staffer)
+
+    print(f"== Extracted {len(csv_staffers)} staffers")
+    return csv_staffers
+
+def write_to_db(csv, ricoh_db): # Connects to SQLite db, or creates it if it doesn't exist
+
+    def  _insert_staffer(row, conn):
+        c = conn.cursor()
+        c.execute(f"""INSERT INTO '{debug_datestring}log' (user,name,bnwprints,colorprints)
+                      VALUES ('{row['num']}','{row['name']}','{row['b&wtotal']}','{row['colortotal']}')""")
+
+    conn = None
+    
+    try:
+        conn = sqlite3.connect(ricoh_db)
+        print(sqlite3.version)
+    except Error as e:
+        print(e)
+
+    try:
+        c = conn.cursor()
+        c.execute(f"""CREATE TABLE IF NOT EXISTS '{debug_datestring}log' (
```

Comment:
Per my email to you, I think we really only want to have a single table.  The time/date stamp should be one of the entries in the table.  Think of it as a giant spreadsheet with several columns:

* Unique row identifier
* Time/datestamp
* Username
* Department
* BW prints
* Color prints
* ...any other relevant data

It may actually be worth parsing out *all* the numeric values from the CSV and saving them all in the DB.  We may only make reports with the total B&W and color values, but having *all* the data could be helpful someday when we come up with a new question that we want to answer.  I.e., it would be horrible if someday we said "Well, we could have answered that question if we had saved all the data over time, but now we don't have that data any more..."

Reply:
Also, I'm not sure "prints" is the right column.  I think "prints" may be what people print from their computers, but it doesn't cover what they _copy_ (i.e., physically stand at the Richo and make copies).  I seem to recall that there's a summation column that's the total of _all_ B&W and _all_ color generations -- I think that that is what we're ultimately going to want to report.

But per my prior comment, I think saving *all* the values from the CSV into the database is probably the right thing to do.  I.e., we should save the summation, the prints, the copies, ...etc.  Everything.